internal/controllers/user_devices_controller.go

```diff
@@ -616,29 +616,56 @@ func (udc *UserDevicesController) RegisterDeviceForUserFromVIN(c *fiber.Ctx) err
 
 	// request valuation
 	if udc.Settings.IsProduction() {
-		message := services.ValuationDecodeCommand{
-			VIN:          vin,
-			UserDeviceID: udFull.ID,
-		}
-		messageBytes, err := json.Marshal(message)
-
-		if err != nil {
-			localLog.Err(err).Msg("Failed to marshal message.")
-		} else {
-			pubAck, err := udc.NATSSvc.JetStream.Publish(udc.NATSSvc.JetStreamSubject, messageBytes)
-			if err != nil {
-				localLog.Err(err).Msg("failed to publish to NATS")
-			} else {
-				localLog.Info().Str("user_device_id", udFull.ID).Msgf("published valuation request to NATS with Ack: %+v", pubAck)
-			}
-		}
+		udc.requestValuation(vin, udFull.ID)
+		udc.requestInstantOffer(udFull.ID)
 	}
 
 	return c.Status(fiber.StatusCreated).JSON(fiber.Map{
 		"userDevice": udFull,
 	})
 }
 
+func (udc *UserDevicesController) requestValuation(vin string, userDeviceID string) {
+	message := services.ValuationDecodeCommand{
```

Comment:
while we're here, let's also add the tokenId since we'll need it for future when we migrate valuations-api workers.